pytorch_lightning/core/__init__.py

```diff
@@ -13,6 +13,6 @@
 # limitations under the License.
 
 from pytorch_lightning.core.datamodule import LightningDataModule
-from pytorch_lightning.core.lightning import LightningModule
+from pytorch_lightning.core.module import LightningModule
```

Comment:
You need to keep the pytorch_lightning.core.lightning module. 
Otherwise existing imports break and this is too big of a breaking change.
Right now, this fails with a ModuleNotFoundError:
`from pytorch_lightning.core.lightning import LightningModule`

Reply:
will be done in a follow-up.
It might break on master for some time though, but this was done to avoid huge diff.

Reply:
I just thought that we could do it all here and instead "rebase and merge" the PR. (just proposed it in https://github.com/PyTorchLightning/pytorch-lightning/pull/13031#discussion_r870574737)